File: 알고리즘/1/해시/위장/위장.py

def solution(clothes):
    data = dict()

    for clothe in clothes:
        data[clothe[1]] = data.get(clothe[1], 0) + 1

    answer = 1
    for ele in data.values():
        answer *= (ele+1)
    print(answer)
    answer -= 1
    return answer

File: algorithms/1/해시/위장/위장.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<vector<string>> clothes)
{
    int answer = 1;
    unordered_map<string, int> data;
    for (auto clothe : clothes)
    {
        data[clothe[1]] += 1;
    }

    for (auto it = data.begin(); it != data.end(); it++)
    {
        answer *= it->second + 1;
    }
    answer-=1;
    return answer;
}